Replace got with fetch

wesleytodd opened the pull request:
While the tests pass and it all seems fine, the `cache` key used to be a `Map` for `got` to use. Now it is just a simple in memory lookup. Same api to pass, but technically someone could have been looking at the contents of the map they passed in, so this could break them so marking it as major.

wesleytodd committed:
feat!: removed got and replaced it with fetch

wesleytodd commented on package.json:
```diff
@@ -34,11 +34,11 @@
     "yargs": "^16.2.0"
   },
   "devDependencies": {
+    "@types/node": "^20.14.11",
```
There was in implicit dependency on these types which was satisfied by bringing in got. Why it had it as a prod dep I don't think I understand, but removing `got` broke the type generation.

ljharb replied:
it should be node 18, given engines.node, no?

RafaelGSS commented:
@wesleytodd can we move on with this? Also, I think this is not a breaking change, is it?

wesleytodd commented:
I am find doing it as minor, and yeah we can move on it, I just have not had time to test and make sure everything was good, nor do the release. Do you have perms do to all that? If not, and I am this delayed again circling back feel free to ping me in slack.

wesleytodd commented:
I am probably unable to make these updates, but I agree with them. If someone wants to take this over feel free.

wesleytodd commented:
It always comes back around. I am going to pick up #34 in the next month and so will try to make these updates and land it to include in that. Let me know if anyone has any other feedback and I will try to incorporate it.

wesleytodd committed:
fix: drop support for <node@20.x update engines to include future versions

wesleytodd commented:
Rebased `1.0.0` on top of `main` and then this on top of that (which was actually kind of a PITA after this long, but I think I got it all correct 🤞). Added a few folks who had either commented or recently helped review some of the stuff going into this tools. Hoping to get the `1.0.0` out sometime next week.
